src/udev/udev-builtin-path_id.c

```diff
@@ -488,12 +498,12 @@ static sd_device *handle_usb(sd_device *parent, char **path) {
 
         if (sd_device_get_sysname(parent, &str) < 0)
                 return parent;
-        port = strchr(str, '-');
-        if (!port)
+
+        hyphen = strchr(str, '-');
+        if (!hyphen)
                 return parent;
-        port++;
 
-        path_prepend(path, "usb-0:%s", port);
+        path_prepend(path, "usb-%.*s:%s", (int)(hyphen - str), str, hyphen + 1);
```

Comment:
This defeats the very purpose of stable device path. USB host number may change across reboots (and probably even without reboot). The part after USB host number is determined by device topology and so does not change.

The actual USB host number here is irrelevant here, because we build stable path to the parent which uniquely identifies USB host. In hindsight it had been better to not include it at all.

What exact problem this change solves?

Reply:
Thank you for the explanation. I completely misunderstand the motivation of using '0' here. I added a comment about that based on your comment.